tests/product/base_product_case.py

```diff
@@ -124,16 +122,25 @@ def setup_docker_cluster(self, cluster_type='centos'):
 
     def tearDown(self):
         self.restore_stdout_stderr_keep_open()
-        self.docker_cluster.tear_down_containers()
-
-    def build_dist_if_necessary(self):
-        if not os.path.exists(DIST_DIR) or not fnmatch.filter(
-                os.listdir(DIST_DIR), 'prestoadmin-*.tar.bz2'):
-            self.docker_cluster.clean_up_presto_test_images()
-            self.build_installer_in_docker()
-        return DIST_DIR
+        if 'docker_cluster' in locals():
+            self.docker_cluster.tear_down_containers()
 
-    def build_installer_in_docker(self):
+    def build_dist_if_necessary(self, cluster=None):
+        if not cluster:
+            cluster = self.docker_cluster
+        if (not cluster.get_dist_dir() or
```

Comment:
It will always be necessary to build the dist directory because the directory names are unique.